src/sidebar/components/ThreadList.tsx

```diff
@@ -212,32 +213,34 @@ export default function ThreadList({ threads }: ThreadListProps) {
     }
   }, [store, newAnnotationTag]);
 
-  const mostRecentHighlightedAnnotationId = useMemo(() => {
-    const highlightedAnnos = allAnnotations.filter(
-      anno => anno.id && highlightedAnnotations.includes(anno.id),
-    );
-    // Get the annotation with most recent updated field, which contains a
-    // date in ISO format. This means their alphabetical and chronological
-    // orders match.
-    const mostRecentHighlightedAnnotation =
-      highlightedAnnos.reduce<Annotation | null>(
-        (mostRecent, current) =>
-          !mostRecent || mostRecent.updated < current.updated
-            ? current
-            : mostRecent,
-        null,
-      );
-
-    return mostRecentHighlightedAnnotation?.id;
-  }, [allAnnotations, highlightedAnnotations]);
-
-  // Scroll to the most recently highlighted annotation, unless creating/editing
+  const mostRelevantHighlightedAnnotationId = useMemo(() => {
+    // To determine which is the most relevant annotation, we first sort them
+    // from newest to oldest
+    const sortedHighlightedAnnos = allAnnotations
+      .filter(anno => anno.id && highlightedAnnotations.includes(anno.id))
+      // The `updated` field contains a date in ISO format. This means their
+      // alphabetical and chronological orders match.
+      .sort((a, b) => b.updated.localeCompare(a.updated));
+
+    // Then we find the most recent one with mentions of current user, or simply
+    // the most recent one
+    const mostRelevantHighlightedAnnotation =
+      sortedHighlightedAnnos.find(
+        anno =>
+          anno.mentions &&
+          anno.mentions.some(({ userid }) => userid === currentUserId),
+      ) ?? sortedHighlightedAnnos[0];
+
+    return mostRelevantHighlightedAnnotation?.id;
+  }, [allAnnotations, currentUserId, highlightedAnnotations]);
```

Comment:
This is another case where I think it will make testing and debugging easier if you split out the logic for finding the annotation into a separately tested function. Then for `ThreadList` you only need the comparatively simpler test that the client scrolls to the selected annotation ID.

Reply:
Yeah, you are right.